### ISSUE TITLE
Pinned requests conflicts with jupyterlab-server.

### ISSUE THREAD

**Issue by picbeats** [pasqal-io/pasqal-cloud#58]
pinned requests conflicts with jupyterlab-server
* jupyterlab-server==2.19.0
 - requests [required: >=2.28, installed: 2.25.1]
or 
* pasqal-sdk==0.1.6
 - requests [required: ==2.25.1, installed: 2.28.2]

Is pinning of request-mock is also required? 
Is request-mock is a functional requirements or needed only for testing?

[offtopic] pulser-pasqual=0.9.0 pins pasqal-sdk=0.1.6 version, have you considered to use semantic versioning rather than pinning, so pasqal-sdk could be upgraded independently?

**Comment by MatthieuMoreau0** [comment]
Hello @picbeats,

Sorry for the very late reply. I'll try to answer your questions:
- I have pushed an [MR](https://github.com/pasqal-io/pasqal-cloud/pull/77) to unpin the python dependencies as you suggested, let me know if that looks good to you.  
- `request-mock` has been moved to dev dependencies in a previous MR as it is only required for tests. I assume I can keep that version pinned as you will not install it to use the pasqal_cloud SDK.
- We are indeed moving towards semantic versioning which should make the package upgrades less painful ( https://github.com/pasqal-io/pasqal-cloud/issues/73)

Let me know if you have other questions!
